test/unit/controllers/DeclarationControllerSpec.scala

```diff
@@ -119,15 +123,46 @@ class DeclarationControllerSpec extends ControllerSpecBase with BeforeAndAfterEa
     "return OK and the correct view for a POST" in {
       givenTheCaseCreatesSuccessfully()
       givenTheAttachmentPublishSucceeds()
+      givenTheCaseCreatedEventIsSuccessful
 
       val result = await(controller(extractDataFromCache).onSubmit(NormalMode)(fakeRequest))
 
       result.header.status shouldBe Status.SEE_OTHER
       result.header.headers(LOCATION) shouldBe "/foo"
     }
 
+    "create an event when the ATAR application has been submitted successfully" in {
+      givenTheCaseCreatesSuccessfully()
+      givenTheCaseCreatedEventIsSuccessful()
+
+      val c = controller(extractDataFromCache)
+      val req = fakeRequest
+
+      await(c.onSubmit(NormalMode)(req))
+
+      verify(casesService, times(1)).addCaseCreatedEvent(
+        any[Case], any[Operator])(any[HeaderCarrier])
```

Comment:
I don't think we have actually tested that we are creating the event with correct fields.

If we define a contact name,
```
when(contact.name).thenReturn("luigi")
```

then instead of `any[Case]` and `any[Operator]` we can ensure it is using the correct fields.

```suggestion
        refEq(createdCase), refEq(Operator("", Some("luigi"))))(any[HeaderCarrier])
```